pyglmnet/pyglmnet.py

```diff
@@ -472,12 +472,17 @@ class GLM(BaseEstimator):
         learning rate for gradient descent.
         default: 2e-1
     max_iter: int
-        maximum iterations for the model.
+        maximum number of iterations for the solver.
         default: 1000
     tol: float
         convergence threshold or stopping criteria.
-        Optimization loop will stop when norm(gradient) is below the threshold.
+        Optimization loop will stop when relative change
+        in parameter norm is below the threshold.
         default: 1e-6
+    htol: float
+        tolerance for Hessian term. if the Hessian is below this value
+        the learning rate will be used as step size.
+        default: 1e-3
```

Comment:
as we discussed privately, we don't need to expose it for now. If for some reason we realize that users need to tweak it, we can always expose it later. However, "unexposing" is harder because you will need to have a deprecation cycle then. We can have the `htol` declared as a global variable so that it's not a dangling constant in the code.

Reply:
i've simplified it even further. we simply do `1. / (1. + hk)` instead of `1. / hk`: it's a special case of Quasi Newton for 1 dimensional Newton update for a convex problem